src/harness/testcases/WINNF_FT_S_FAD_testcase.py

```diff
@@ -0,0 +1,263 @@
+                                         sas_test_harness_dump_records)
+    # Start the server
+    sas_test_harness.start()
+
+    # Whitelist the FCCID and UserID for CBSD C2 in SAS UUT
+    self._sas_admin.InjectFccId({'fccId': device_c2['fccId']})
+    self._sas_admin.InjectUserId({'userId': device_c2['userId']})
+
+    # Whitelist the FCCID and UserID for CBSD C4 in SAS UUT
+    self._sas_admin.InjectFccId({'fccId': device_c4['fccId']})
+    self._sas_admin.InjectUserId({'userId': device_c4['userId']})
+
+    # Whitelist the FCCID and UserID of the devices loaded to SAS-test_harness in SAS UUT
+    for cbsdRecord in sas_test_harness_dump_records_config['cbsdRecords']:
+      self._sas_admin.InjectFccId({'fccId': cbsdRecord['registration']['fccId']})
+      self._sas_admin.InjectUserId({'userId': cbsdRecord['registration']['userId']})
+
+    # Send a valid Registration Request for CBSD (C2) to the SAS UUT
+    # Check registration response of SAS UUT
+    # Send a valid Grant Request Message for CBSD (G2) to the SAS UUT
+    # Check grant response of G2 from SAS UUT
+
+    cbsd_ids_c2, grant_ids_c2 = self.assertRegisteredAndGranted([device_c2], [grant_g2])
+
+    # Send a valid Registration Request for CBSD (C4) to the SAS UUT
+    # Check registration response  of SAS UUT
+    # Send a valid Grant Request for CBSD (G4) to SAS UUT
+    # Check grant response of G4 from SAS UUT
+
+    cbsd_ids_c4, grant_ids_c4 = self.assertRegisteredAndGranted([device_c4], [grant_g4])
+
+    # Send the Heartbeat request for the Grant G2 and G4 of CBSD C2 and C4
+    #                                    to SAS UUT
+    transmit_expire_times = self.assertHeartbeatsSuccessful(
+        list([cbsd_ids_c2, cbsd_ids_c4]),
+        list([grant_ids_c2, grant_ids_c4]),
+        list([[('GRANTED')], [('GRANTED')]])
+    )
+
+    # Notify the SAS UUT about the SAS Test Harness
+    certificate_hash = getCertificateFingerprint(sas_test_harness_config['serverCert'])
+    self._sas_admin.InjectPeerSas({'certificateHash': certificate_hash,
+                                   'url': sas_harness_base_url})
+
+    # Trigger CPAS in the SAS UUT and wait until complete.
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+    # Send the Heartbeat request again for the G2 and G4 to SAS UUT
+    request = {
+        'heartbeatRequest': [
+            {
+                'cbsdId': cbsd_ids_c2[0],
+                'grantId': grant_ids_c2[0],
+                'operationState': 'GRANTED'
+            },
+            {
+                'cbsdId': cbsd_ids_c4[0],
+                'grantId': grant_ids_c4[0],
+                'operationState': 'GRANTED'
+            }
+        ]}
+    responses = self._sas.Heartbeat(request)['heartbeatResponse']
+    for response in responses:
+      self.assertTrue(response['response']['responseCode'] in (103, 500))
+
+    sas_test_harness.shutdown()
+    del sas_test_harness
```

Comment:
When   we tested the code we found that the python garbage collector not very   consistent. We didn't see the directory getting deleted. However if we call   this explicitly the  clean up happens   properly. Hence we would like to retain this code. Hope this is fine.

Reply:
I'm glad you looked into it so thoroughly! Please add a note with an explanation so that someone doesn't "help" you later by deleting this line :)

Reply:
Please add more detail to the comment -- like the detail you gave in your GitHub comment. The current comment is too generic to prevent someone from "helping" later.